plugins/usd/module/vtkF3DUSDImporter.cxx

```diff
@@ -0,0 +1,966 @@
+
+    auto& tex = this->TextureMap[samplerPrim.GetPath().GetAsString()];
+
+    if (tex == nullptr)
+    {
+      pxr::TfToken idToken;
+      bool defined = samplerPrim.GetIdAttr().Get(&idToken);
+      if (!defined || idToken != pxr::TfToken("UsdUVTexture"))
+      {
+        // only pxr::UsdUVTexture supported for now
+        return nullptr;
+      }
+
+      pxr::SdfAssetPath path;
+      if (samplerPrim.GetInput(pxr::TfToken("file")).Get(&path))
+      {
+        vtkSmartPointer<vtkImageReader2> reader;
+
+// CreateImageReader2FromExtension needs https://gitlab.kitware.com/vtk/vtk/-/merge_requests/8211
+#if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 0, 20210729)
+        const std::string& assetPath = path.GetAssetPath();
+        std::string ext = assetPath.substr(assetPath.find_last_of('.'));
+        reader.TakeReference(vtkImageReader2Factory::CreateImageReader2FromExtension(ext.c_str()));
+#endif
+
+        if (!reader)
+        {
+          // cannot read the image file
+          return nullptr;
+        }
+
+        const std::string& resolvedPath = path.GetResolvedPath();
+        auto asset = pxr::ArGetResolver().OpenAsset(pxr::ArResolvedPath(resolvedPath));
+
+        if (!asset)
+        {
+          // cannot get USD asset
+          return nullptr;
+        }
+
+        auto buffer = asset->GetBuffer();
+
+        if (!buffer)
+        {
+          // buffer invalid
+          return nullptr;
+        }
+
+        reader->SetMemoryBuffer(buffer.get());
+        reader->SetMemoryBufferLength(asset->GetSize());
+        reader->Update();
+
+        tex = reader->GetOutput();
+      }
+    }
+
+    // extract component based on token
+    vtkNew<vtkImageExtractComponents> extract;
+    extract->SetInputData(tex);
+
+    const std::string& channels = token.GetString();
+
+    if (channels == "rgb")
+    {
+      extract->SetComponents(0, 1, 2);
+    }
+    else if (channels == "r")
+    {
+      extract->SetComponents(0);
+    }
+    else if (channels == "g")
+    {
+      extract->SetComponents(1);
+    }
+    else if (channels == "b")
+    {
+      extract->SetComponents(2);
+    }
+    else if (channels == "a")
+    {
+      extract->SetComponents(3);
+    }
+    else
+    {
+      // just return the image as is
+      return tex;
+    }
+
+    extract->Update();
+    return extract->GetOutput();
+  }
+
+  std::pair<vtkSmartPointer<vtkProperty>, bool> GetVTKProperty(
+    const pxr::UsdShadeShader& shaderPrim)
+  {
+    auto& prop = this->ShaderMap[shaderPrim.GetPath().GetAsString()];
+
+    bool isTranslucent = false;
+
+    if (prop == nullptr)
+    {
+      pxr::UsdAttribute attr = shaderPrim.GetIdAttr();
+
+      if (attr)
+      {
+        pxr::TfToken materialToken;
+
+        bool defined = attr.Get(&materialToken);
+
+        if (!defined || materialToken != pxr::TfToken("UsdPreviewSurface"))
+        {
+          // only pxr::UsdPreviewSurface supported for now
+          return { nullptr, isTranslucent };
+        }
```

Comment:
same, should be testable

Reply:
Couldn't find an asset with a different material token.